_docs-sources/iac/usage/using-the-library.md

```diff
@@ -0,0 +1,396 @@
+	"os"
+	"testing"
+
+	"fmt"
+
+	awsSDK "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/lambda"
+	"github.com/gruntwork-io/terratest/modules/aws"
+	"github.com/gruntwork-io/terratest/modules/random"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestLambdaCreated(t *testing.T) {
+	// Run this test in parallel with all the others
+	t.Parallel()
+
+	// Unique ID to namespace resources
+	uniqueId := random.UniqueId()
+	// Generate a unique name for each Lambda so any tests running in parallel don’t clash
+	lambdaName := fmt.Sprintf("test-lambda-%s", uniqueId)
+
+	// Get the cwd so we can point to the lambda handler
+	path, _ := os.Getwd()
+	srcPath := path + "/src"
+
+	terraformOptions := &terraform.Options{
+		// Where the Terraform code is located
+		TerraformDir: "../serverless-api/lambda/",
+
+		// Variables to pass to the Terraform code
+		Vars: map[string]interface{}{
+			"name":        lambdaName,
+			"runtime":     "python3.9",
+			"handler":     "main.lambda_handler",
+			"source_path": srcPath,
+		},
+	}
+
+	// Run 'terraform destroy' at the end of the test to clean up
+	defer terraform.Destroy(t, terraformOptions)
+
+	// Run 'terraform init' and 'terraform apply' to deploy the module
+	terraform.InitAndApply(t, terraformOptions)
+
+	// Create a lambda client so we can retrieve the function
+	lambdaClient := aws.NewLambdaClient(t, "us-west-2")
+	function, _ := lambdaClient.GetFunction(&lambda.GetFunctionInput{
+		FunctionName: &lambdaName,
+	})
+
+	// Assert the function name is equal to what we set
+	assert.Equal(t, lambdaName, awsSDK.StringValue(function.Configuration.FunctionName))
+}
+```
+
+In this test, we first generate data so that the test run creates resources with unique names. Next, we create the Terraform `options`, which indicates the folder in which the Terraform module we want to test is located and sets the values that will be passed in for variables. Then, we set up a `terraform destroy` operation, which will always run regardless of the test status. Then, we run `terraform init` and `terraform apply` to create the resources. Finally, we validate that the name of the AWS Lambda function that was created matches the expected name.
+
+### Run the test
+
+Finally, run the test you wrote. From the `test` directory, run the following command:
+```bash
+go test -v
+```
+
+You should expect to see `--- PASS: TestLambdaCreated` in the final log lines of the output from the test.
+
+## What’s next
+
+Now that you’ve used a Gruntwork module to provision resources, consider how you would need to expand this usage make the Lambda function available via a URL using an [AWS API Gateway HTTP API](../../reference/modules/terraform-aws-lambda/api-gateway-proxy/). We refer to combining multiple modules into a single deliverable module as a [service](../overview/services.md).
+
+<!-- Leaving this commented out since these two docs pages are coming soon -->
```

Comment:
```suggestion
```

Reply:
Removed in https://github.com/gruntwork-io/docs/pull/838#:~:text=Verified-,5d6ac59,-Merge%20state